acvm/src/pwg/directives/mod.rs

```diff
@@ -58,7 +58,7 @@ fn solve_directives_internal(
                 None => FieldElement::one(),
             };
 
-            let (int_r, int_q) = if pred_value.is_zero() {
+            let (int_r, int_q) = if pred_value.is_zero() || int_b.is_zero() {
```

Comment:
```suggestion
            let (int_r, int_q) = if pred_value.is_zero() || (int_a.is_zero() && int_b.is_zero()) {
```
Don't we want this? This will give us (0, 0) as the remainder and quotient for whenever int_b is zero. We should add a test for when divisor_is_zero but the dividend is not zero.